lib/chef/provider.rb

```diff
@@ -200,12 +310,14 @@ def recipe_eval(&block)
       # manipulating notifies.
 
       converge_by ("evaluate block and run any associated actions") do
-        saved_run_context = @run_context
-        @run_context = @run_context.dup
-        @run_context.resource_collection = Chef::ResourceCollection.new
-        instance_eval(&block)
-        Chef::Runner.new(@run_context).converge
-        @run_context = saved_run_context
+        saved_run_context = run_context
+        begin
+          @run_context = run_context.create_child
```

Comment:
Does `create_child` create an empty resource collection or does it keep all the existing resources?